src/awkward/_connect/numpy.py

```diff
@@ -207,23 +207,29 @@ def _array_ufunc_adjust_apply(
         )
 
 
-def _array_ufunc_signature(ufunc, inputs):
-    signature = [ufunc]
+def _array_ufunc_signature(ufunc, inputs) -> tuple[Any, ...] | None:
+    signature = []
+    has_seen_nominal_type = False
     for x in inputs:
         if isinstance(x, ak.contents.Content):
             record_name, list_name = x.parameter("__record__"), x.parameter("__list__")
             if record_name is not None:
                 signature.append(record_name)
+                has_seen_nominal_type = True
             elif list_name is not None:
                 signature.append(list_name)
+                has_seen_nominal_type = True
             elif isinstance(x, NumpyArray):
                 signature.append(x.dtype.type)
             else:
                 signature.append(None)
         else:
             signature.append(type(x))
 
-    return tuple(signature)
+    if has_seen_nominal_type:
+        return (ufunc, *signature)
+    else:
+        return None
```

Comment:
A signature must have at least one nominal type in it (at least one string). We should not be allowing overrides of functions based on types that are not named by parameters (`__array__`, `__record__`, or `__list__`).

We talked about this on our call, and I think it's a good rule. (I hadn't, previously, intended the alternative: overriding ufunc behaviors when there are no named types in the Awkward Array.)

Good! :heavy_check_mark: